deploy-service/common/src/main/java/com/pinterest/deployservice/common/HTTPClient.java

```diff
@@ -41,33 +41,51 @@ private String generateUrlAndQuery(String url, Map<String, String> params) throw
         for (Map.Entry<String, String> entry : params.entrySet()) {
             sb.append(prefix);
             prefix = "&";
+            // note:  scrubUrlQueryValue could be expensive with many filtered values
+            //        consider using it only in only a DEBUG logging context in the future
+            String reportedValue = scrubUrl ? scrubUrlQueryValue(entry.getKey(), entry.getValue()) : entry.getValue();
             sb.append(String.format("%s=%s", entry.getKey(),
-                URLEncoder.encode(entry.getValue(), "UTF-8")));
+                URLEncoder.encode(reportedValue, "UTF-8")));
         }
         return sb.toString();
     }
 
+    public String scrubUrlQueryValue(String queryParamKey, String queryParamValue) {
```

Comment:
Temporarily converted to public method to discuss testing practices (see unit tests file).